src/app/reporting/Engine.cpp

```diff
@@ -277,58 +266,62 @@ CHIP_ERROR Engine::BuildSingleReportDataEventReports(ReportDataMessage::Builder
     {
         ExitNow(); // Read clean, move along
     }
-
-    while (apReadHandler->GetCurrentPriority() != PriorityLevel::Invalid)
+    else
```

Comment:
This is an early return, why we need another `else`

Reply:
Yep, would remove, thanks